File: source/algorithm/NQ029/NQ029_improved.py
def is_palindrome(x):
    if x < 0:
        return False
    res, oldx = 0, x
    while x:
        res = res * 10 + x % 10
        x //= 10
    return res == oldx

def main():
    import sys
    for line in sys.stdin:
        n = int(line.strip())
        print("true" if is_palindrome(n) else "false")

if __name__ == "__main__":
    main()

File: 兴趣班入门百练讲义/NQ100/NQ029/NQ029_improved.cpp
#include <iostream>
using namespace std;

bool isPalindrome(int x) {
    if (x < 0) return false;
    long long res = 0, oldx = x;
    while (x) {
        res = res * 10 + x % 10;
        x /= 10;
    }
    return res == oldx;
}

int main() {
    int n;
    while (cin >> n) {
        cout << (isPalindrome(n) ? "true" : "false") << endl;
    }
    return 0;
}